inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp

```diff
@@ -219,11 +219,20 @@ InferenceEngine::Blob::Ptr MKLDNNPlugin::MKLDNNInferRequest::GetBlob(const std::
         if (it != _preProcData.end()) {
             data = it->second->getRoiBlob();
             return data;
-        }
-
-        if (_inputs.find(name) != _inputs.end()) {
+        } else if (_inputs.find(name) != _inputs.end()) {
             data = _inputs[name];
             checkBlob(data, name, true);
+            // check if preprocess required, but still wasn't set
+            auto preProcess = std::find_if(std::begin(_networkInputs), std::end(_networkInputs),
+                                           [&](const std::pair<std::string, InferenceEngine::InputInfo::Ptr>& pair) {
+                                               return pair.first == name; })->second->getPreProcess();
+            // we don't need to check resize preprocessing here,
+            // because using of resize supposes an explicit setting of blob in SetBlob() method
```

Comment:
What if we call getblob and reset blob every time after with new different size? In this case we can use resize with getblob

Reply:
Right. Corrected